pkg/component/base/execution_wrapper.go

```diff
@@ -108,6 +108,14 @@ func (e *ExecutionWrapper) Execute(ctx context.Context, jobs []*Job) (err error)
 	for batchIdx, job := range wrappedJobs {
 		outputs[batchIdx] = job.Output.(*outputWriter).GetOutput()
 	}
+
+	// Universal AI component set usesInstillCredentials when it executes, which is different from other AI components.
+	h, err = newUH(e)
```

Comment:
❓ Why is this `h` different than the one declared above?

Reply:
Ah, maybe the comment is misleading.

Because Universal AI component set usesInstillCredentials when it executes, I have to re-initialise the h after the job executes.

So, they are same, and I just re-initialise it.
I will change the comment

Reply:
I don't understand why this needs to be computed during `Execute` instead of `CreateExecution`. Can you expand on this?

Reply:
> why this needs to be computed during Execute instead of CreateExecution.

Because it will need to know which vendor to be called when the input is inserted, which is in the process of Execute.

> Having a component that changes this under the hood and forcing the interface it implements to adapt to that fact feels like an antipattern.

I do not get this point.
Do you mean that it will be better pattern to initialise another handler?
Or, maybe we should differentiate the handler to calculate the input token and the output cost.
Now, it is 1 handler. Check is more like before the Execute.
But, Collect is more like after the Execute because the output usually comes after the execution.
Is this point you are talking about?

```
// UsageHandler allows the component execution wrapper to add checks and
// collect usage metrics around a component execution.
type UsageHandler interface {
	Check(ctx context.Context, inputs []*structpb.Struct) error
	Collect(ctx context.Context, inputs, outputs []*structpb.Struct) error
}
```

Reply:
@chuang8511 
What if we put the model name into setup?

Reply:
The issue arises because we can only resolve secrets in `setup`, not in `input`.
Maybe we should reconsider which data belongs to `setup` and which data belongs to `input`.

Reply:
> The issue arises because we can only resolve secrets in setup, not in input.

Why do we have this limitation?

> Maybe we should reconsider which data belongs to setup and which data belongs to input.

Yes. I think so. The model is a bit weird for me to set it in setup.
But, it seems a good solution too.

Reply:
Although it may not be consistent with other components, placing it in `setup` is still valid. The original purpose of having separate `setup` and `input` sections is to distinguish between 1. how we configure or connect the component and 2. the data fed into it. We can consider `model_name` as a configuration rather than part of the input data.

Reply:
> But recreating the usage handler because one implementation changes it during execution is obscure, it can lead to hard-to-debug issues.

It is a good opinion.

> I'd vote for keeping this in the setup but I'm open for alternatives as long as the interface is clear and dictates the implementations, not the other way around.

> Although it may not be consistent with other components, placing it in setup is still valid.

Ok, I will modify it. Thanks for feedback.

Reply:
I move it to setup.